core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java

```diff
@@ -778,17 +781,36 @@ private static Object getOrElseProperty(CamelContext context, Object target, Str
 
         // use configurer if possible
         Object answer = null;
-        GeneratedPropertyConfigurer configurer = context.adapt(ExtendedCamelContext.class).getConfigurerResolver().resolvePropertyConfigurer(target.getClass().getSimpleName(), context);
+        Class<?> type = null;
+
+        GeneratedPropertyConfigurer configurer = PropertyConfigurerHelper.resolvePropertyConfigurer(context, target);
         if (configurer instanceof PropertyConfigurerGetter) {
-            answer = ((PropertyConfigurerGetter) configurer).getOptionValue(target, key, ignoreCase);
+            type = (Class<?>)((PropertyConfigurerGetter)configurer).getAllOptions(target).get(key);
```

Comment:
Maybe this can be done later as you override type in case answer != null and other bits. So wonder if this can be moved down, and to get the type only if you have not computed it